### PR TITLE
feat(hardware) add a python driver for the rear panel mcu

### PR DESCRIPTION
**Pull request by ryanthecoder**
# Overview
Functioning just like the CanMessenger this new driver uses the underling serial port comms to communicate over USB to the rear-panel

# Test Plan

# Changelog

# Review requests

# Risk assessment

### CODE REVIEW

File: hardware/opentrons_hardware/drivers/binary_usb/binary_messenger.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
+"""Can messenger class."""
```
Comment (sfoster1) [inline hardware/opentrons_hardware/drivers/binary_usb/binary_messenger.py]:
```suggestion
"""Binary USB messenger class."""
```

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Looks good but let's change that comment.

Also add a test for the async-read interface in `bin_serial`

File: hardware/opentrons_hardware/drivers/binary_usb/bin_serial.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 import logging
 import concurrent.futures
 
-from typing import Optional, Type
+from typing import Optional, Type  # , Any
```
Comment (sfoster1) [inline hardware/opentrons_hardware/drivers/binary_usb/bin_serial.py]:
```suggestion
from typing import Optional, Type
```

File: hardware/opentrons_hardware/drivers/binary_usb/bin_serial.py [hunk 3]
```diff
@@ -84,6 +84,8 @@ async def read(self) -> Optional[BinaryMessageDefinition]:
             header_data = await self._loop.run_in_executor(
                 self._executor, partial(self._port.read, size=4)
             )  # read the message id and length
+            if len(header_data) < 4:
+                return None
```
Comment (sfoster1) [inline hardware/opentrons_hardware/drivers/binary_usb/bin_serial.py:88]: does this happen often? when it does, is it an incomplete read or garbage data?

File: hardware/opentrons_hardware/drivers/binary_usb/binary_messenger.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+        listener: BinaryMessageListenerCallback,
+        filter: Optional[BinaryMessageListenerCallbackFilter] = None,
+    ) -> None:
+        """Add a message listener."""
+        self._listeners[listener] = listener, filter
+
+    def remove_listener(self, listener: BinaryMessageListenerCallback) -> None:
+        """Remove a message listener."""
+        if listener in self._listeners:
+            del self._listeners[listener]
```
Comment (sfoster1) [inline hardware/opentrons_hardware/drivers/binary_usb/binary_messenger.py]: let's make this `self._listeners.remove(listener)` or something, don't love `del` in this case

**Review by sfoster1 — APPROVED** [review]
Approving because it looks good in structure but let's fix some last little nitpicks before merging

File: hardware/opentrons_hardware/drivers/binary_usb/bin_serial.py (hunk 3 above)
Comment (ryanthecoder) [reply hardware/opentrons_hardware/drivers/binary_usb/bin_serial.py:88]: I hadn't seen it happen but I figured its a good guard against random noise or something on the line

**Pull request merged**